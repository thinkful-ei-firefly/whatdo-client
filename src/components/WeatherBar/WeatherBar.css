.weatherbar {
    text-align: center;
}

.weatherbar-header {
    text-align: center;
}

.weathericon {
    display: inline-block;
    margin: 5px;
}

.weathericon-time {
    text-align: center;
    margin: 0;
}

.weathericon-icon {
    margin-top: -20px;
    margin-bottom: -20px;
}

.weathericon-temperature {
    text-align: center;
    margin: 0px;
}

.high {
    color: darkred
}

.low {
    color: darkblue;
}