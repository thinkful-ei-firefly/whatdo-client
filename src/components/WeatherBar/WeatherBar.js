import React from 'react'
import WeatherApiService from '../../services/weather-api-service'
import weatherDataHelper from '../../helpers/weather-data-helpers'
import EventApiService from '../../services/event-api-service'
import eventDataHelpers from '../../helpers/event-data-helpers'

export default class WeatherBar extends React.Component{
  handleButtonClickWeather = async (ev) => {
    ev.preventDefault()
    try {
      const jsonData = await WeatherApiService.weatherReport('30084')
      console.log(jsonData)
      const weatherReport = weatherDataHelper.parseWeatherReport(jsonData)
      console.log(weatherReport)

    } catch(err) {
      console.log(err)
    }
    
  }

  handleButtonClickEvents = async (ev) => {
    ev.preventDefault()
    try {
      const jsonData = await EventApiService.eventSearch('30084')
      console.log(jsonData)
      const eventList = eventDataHelpers.parseEventList(jsonData)
      console.log(eventList)

    } catch(err) {
      console.log(err)
    }
    
  }

  render(){
    return(
      <div>
        <button type="button" onClick={ev => this.handleButtonClickEvents(ev)}>Click me!</button>
        <button type="button" onClick={() => console.log(eventDataHelpers.todayStr())}>DateString</button>
      </div>
    )
  }
}