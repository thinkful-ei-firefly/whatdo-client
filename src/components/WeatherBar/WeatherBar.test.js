import React from 'react';
import ReactDOM from 'react-dom';
import WeatherBar from './WeatherBar';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<WeatherBar />, div);
  ReactDOM.unmountComponentAtNode(div);
});
