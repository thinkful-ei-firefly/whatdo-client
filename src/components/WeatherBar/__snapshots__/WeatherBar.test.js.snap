// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherBar /> renders properly 1`] = `
<div
  className="weatherbar"
  id="scroll-target"
>
  <h2>
    No Weather data available for 
    2019-11-05
  </h2>
</div>
`;
